lib/src/main/java/io/ably/lib/http/CallbackfulHttp.java

```diff
@@ -0,0 +1,423 @@
+package io.ably.lib.http;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.ably.lib.http.Http.RequestBody;
+import io.ably.lib.http.Http.ResponseHandler;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Callback;
+import io.ably.lib.types.ErrorInfo;
+import io.ably.lib.types.Param;
+import io.ably.lib.util.Log;
+
+/**
+ * Created by tcard on 31/8/17.
+ */
+
+public class CallbackfulHttp<Executor extends java.util.concurrent.Executor> {
```

Comment:
What is `CallbackfulHttp` i.e. I am not clear on what the `ful` after `Callback` implies.  Perhaps a high level comment on what this does is needed.